src/isomorphic/hooks/ReactComponentTreeHook.js

```diff
@@ -184,9 +232,14 @@ var ReactComponentTreeHook = {
       // got a chance to mount, but it still gets an unmounting event during
       // the error boundary cleanup.
       item.isMounted = false;
+      if (item.parentID === 0) {
+        var indexInRootIDs = rootIDs.indexOf(id);
```

Comment:
Can we do something that's not O(n^2) to unmount n roots? If getRootIDs is only used in tests than the slow implementation you had would be better.